// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

edition = "2023";

package brain.evolution;

option features.utf8_validation = NONE;

message PopulationClientSpec {
  // Which server to connect to.
  string connect_to_server_id = 1;
  double rpc_deadline_secs = 2;  // Required.

  // Retry parameters.
  double min_delay_secs = 3;  // Required.

  double max_delay_secs = 4;  // Required.

  int64 max_retries = 5;  // Required.

  int64 max_missing_associated_file_retries = 7 [default = 10];

  int64 max_clearing_associated_file_copies_retries = 8 [default = 10];

  double clearing_associated_files_delay_secs = 9 [default = 1];

  // Internal. Do not set.
  int64 worker_collection_id = 10;
  string worker_collection_name = 11;
  string tmp_dir = 12;
}
